pkg/reaper/nodereaper/nodereaper_test.go

```diff
@@ -1391,3 +1409,230 @@ func TestProviderIDParser(t *testing.T) {
 		t.Fatalf("expected Region: %v, got: %v", expectedRegion, providerRegion)
 	}
 }
+
+func TestSkipLabelReaper(t *testing.T) {
+	reaper := newFakeReaperContext()
+	reaper.ReapUnknown = true
+	reaper.ReapUnready = true
+	reaper.AsgValidation = true
+	reaper.FlapCount = 4
+
+	testCase := ReaperUnitTest{
+		TestDescription: "DisableReaper label detection - skip reaping of any node if it has this label",
+		InstanceGroup: FakeASG{
+			Name:      "my-ig.cluster.k8s.local",
+			Healthy:   2,
+			Unhealthy: 0,
+			Desired:   2,
+		},
+		Nodes: []FakeNode{
+			{
+				nodeName:   "node-old",
+				state:      "Ready",
+				ageMinutes: 43100,
+			},
+			{
+				nodeName: "node-flappy",
+				state:    "Ready",
+			},
+			{
+				nodeName:   "node-unknown",
+				state:      "Unknown",
+				lastTransitionMinutes: 6,
+			},
+			{
+				nodeName:   "node-unready",
+				state:      "NotReady",
+				lastTransitionMinutes: 6,
+			},
+		},
+		Events: []FakeEvent{
+			{
+				node:   "node-flappy",
+				count:  4,
+				reason: "NodeReady",
+				kind:   "Node",
+			},
+		},
+		FakeReaper:        	 reaper,
+		ExpectedUnready:   	 2,
+		ExpectedReapable:  	 0,
+		ExpectedDrainable:	 0,
+		ExpectedOldReapable: 0,
+		ExpectedTerminated:	 0,
+		ExpectedDrained:   	 0,
+	}
+	skipLabels := make([]string, len(testCase.Nodes))
+	for i := 0; i < len(testCase.Nodes); i++ {
+		skipLabels[i] = reaperDisableLabelKey
+	}
+	testCase.NodeLabels = skipLabels
+	testCase.Run(t, false)
+}
+
+func TestSkipLabelUnknownNodes(t *testing.T) {
+	reaper := newFakeReaperContext()
+	reaper.ReapUnknown = true
+	reaper.AsgValidation = true
+
+	testCase := ReaperUnitTest{
+		TestDescription: "DisableUnknownReaper label Detection - unknown nodes with skip label are skipped from being reaped",
+		InstanceGroup: FakeASG{
+			Name:      "my-ig.cluster.k8s.local",
+			Healthy:   1,
+			Unhealthy: 0,
+			Desired:   1,
+		},
+		Nodes: []FakeNode{
+			{
+				nodeName:   "node-unknown-1",
+				state:      "Unknown",
+				lastTransitionMinutes: 6,
+			},
+			{
+				nodeName:   "node-unknown-2",
+				state:      "Unknown",
+				lastTransitionMinutes: 6,
+			},
+		},
+		FakeReaper:         reaper,
+		ExpectedUnready:    2,
+		ExpectedReapable:   1,
+		ExpectedDrainable:  0,
+		ExpectedTerminated: 1,
+		ExpectedDrained:    0,
+	}
+	skipLabels := make([]string, len(testCase.Nodes))
+	skipLabels[0] = reapUnknownDisabledLabelKey
+	testCase.NodeLabels = skipLabels
```

Comment:
Sorry. I just noticed this - 
it make more sense to put this under `FakeNode` type.
`createFakeNodes` can make sure the label provided by test is added to the fake nodes.

Example:
``` 
ReaperUnitTest{
		TestDescription: "...",
		InstanceGroup: FakeASG{
			...
		},
		Nodes: []FakeNode{
			{
                                 nodeLabels: map[string]string{reapUnreadyDisabledLabelKey: "true"}
				nodeName:   "node-unknown-1",
				state:      "Unknown",
				lastTransitionMinutes: 6,
			},
		}
}
```

Also, it seems we are not caring about value of the label.. can we add a check in implementation to make sure it's "true"?
The tests can check this is as well if you use a map instead of a slice for NodeLabels (as above)

Reply:
So the implementation actually does check that the label has value "true". Under `hasSkipLabel()` it'll check that the map value is equal to true. If it doesn't have the label and we try to fetch it, it should return an empty string correct? 

But I will change the node labels to a map and move it to the FakeNode struct.

Reply:
Oh right, I see it, it's correct.
👍 
We just need to make sure it's part of the test e.g. by using a map you can also test a label with incorrect value